agent/agent.go

```diff
@@ -1851,8 +1858,14 @@ func (a *Agent) AddCheck(check *structs.HealthCheck, chkType *structs.CheckType,
 			return fmt.Errorf("Check is not valid: %v", err)
 		}
 
-		if chkType.IsScript() && !a.config.EnableScriptChecks {
-			return fmt.Errorf("Scripts are disabled on this agent; to enable, configure 'enable_script_checks' to true")
+		if chkType.IsScript() {
+			if source == ConfigSourceLocal && !a.config.EnableScriptChecks && !a.config.EnableLocalScriptChecks {
```

Comment:
I am not a fan of having several semantics into the code. I would perform the change to only check for a.config.EnableLocalScriptChecks or EnableRemoteScriptChecks

Reply:
:+1: